app/components/newsPage/cards/NewsCollabCommentCard.tsx

```diff
@@ -40,6 +42,9 @@ function NewsCollabCommentCard(props: NewsCollabCommentCardProps) {
     handleDelete,
   } = useNewsCollabCommentCard(props);
 
+  const { filters } = useNewsFeed();
```

Comment:
I wouldn't use the `useNewsFeed` hook in every component. This introduces a tight coupling with the news feed which we don't really want in every card.

Reply:
My suggestions would be to create an extra context that stores the string to be highlighted:

```typescript

interface HighlightContext {
  highlightString?: string;
}

const HighlightContext = createContext<HighlightContext>({});

function HighlightContextProvider(props: PropsWithChildren) {
  const [highlightString, setHighlightString] = useState<string>();
  const { filters } = useNewsFeed();

  useEffect(
    function updateNewsFeedSearchTerm() {
      setHighlightString(filters.searchString);
    },
    [filters.searchString],
  );
  const ctx: HighlightContext = { highlightString };
  return <HighlightContext.Provider value={ctx}>{props.children}</HighlightContext.Provider>;
}

function useHighlightContext() {
  const ctx = useContext(HighlightContext);
  return ctx;
}
```

Then separate the text highlighting into a component (or hook):

```typescript

function HighlightText({ text }: { text: string }) {
  const { highlightString } = useHighlightContext();
  const regex = useMemo(() => new RegExp(`(${highlightString})`, 'gi'), [highlightString]);

  const renderText = (text: string) => {
    const parts = text.split(regex);

    return parts.map((part, index) =>
      part.toLowerCase() === highlightString?.toLowerCase() ? (
        <span key={index} style={{ backgroundColor: 'yellow', fontWeight: 'bold' }}>
          {part}
        </span>
      ) : (
        part
      ),
    );
  };

  return <>{renderText(text)}</>;
}

```

This way, it is easier to implement some other source of highlighting (if we don't want to set the string to be highlighted to the news feed filter, but something else later on)

Usage:

```html

<Box>
  <HighlightText text="Wow a test, isnt it?" />
</Box>

```